core/src/main/java/com/linecorp/armeria/client/DefaultClientRequestContext.java

```diff
@@ -155,6 +172,24 @@ private DefaultClientRequestContext(
         additionalRequestHeaders = options.getOrElse(ClientOption.HTTP_HEADERS, HttpHeaders.of());
     }
 
+    @Nullable
+    private static DefaultServiceRequestContext defaultServiceRequestContext() {
+        final RequestContext current = RequestContext.currentOrNull();
+        if (current instanceof DefaultServiceRequestContext) {
+            return (DefaultServiceRequestContext) current;
+        }
+        return null;
+    }
+
+    @Nullable
+    private static AttributeMap serviceContextAttributeMap(
+            @Nullable DefaultServiceRequestContext serviceContext) {
+        if (serviceContext != null) {
+            return serviceContext.attributeMap();
```

Comment:
Isn't `RequestContext` an `AttributeMap`? I think we can use `ServiceRequestContext` instead of `DefaultServiceRequestContext` in this class.

Reply:
Yes, I didn't have to. 😄 Let me just use `ServiceRequestContext`, thanks!